File: typescript/word-count/word-count.ts
class Words {
    constructor() {}

    count(inp: string): Map<string, number> {
        var wordCount = new Map<string, number>();
        inp.trim().toLowerCase().split(/\s/).filter(x => x.length != 0)
            .forEach(word => wordCount.set(word, ((wordCount.get(word) || 0) + 1) || 1)) // wordCount.get(word) + 1))
        return wordCount
    }
}

export default Words

File: scala/word-count/src/main/scala/WordCount.scala
class WordCount(inp: String) {
  def countWords: Map[String, Int] = {
    inp.toLowerCase.split("[^\\w']+").groupBy(identity).mapValues(_.length)
//    .filter(x => x.isLetterOrDigit || x.isWhitespace || x == '\'' || x == '.' || x == '.')
  }
}


object WordCount {
  def apply(inp: String): WordCount = new WordCount(inp)
}
